### PR TITLE
add list functions to rnrs.ss

### PR DESCRIPTION
**Pull request by jitwit**
* Adds rnrs list functions `filter`, `find`, `assoc`, `member`, `remove`, `remq`, `remp`, and `reverse` to `rnrs.ss`
* Replaces uses of `filter-out` with `remp`
* Add comment about `memp` while parsing variable definitions
* Update readme to reflect added define forms
* Rewrite quadratic `fold-left` with `append` as `fold-right` in `parse-libraries`

I'm supposing it'd be best to add tests for the added functions? One test per function or would it be ok to group them together in something like `test/rnrs-list-functions.ss`?

### CODE REVIEW

File: schism/compiler.ss [hunk 1]
```diff
@@ -529,11 +528,11 @@
...
-            (imported-functions (fold-left (lambda (functions lib)
-                                             (append functions
-                                                     (cdr (parse-library lib imported-envs))))
-                                           '()
-                                           (cdr libs))))
+            (imported-functions (fold-right (lambda (lib functions)
+                                              (append (cdr (parse-library lib imported-envs))
+                                                      functions))
+                                            '()
+                                            (cdr libs))))
```
Comment (jitwit) [inline schism/compiler.ss:535]: I think this was quadratic because the `functions` list gets traversed repeatedly in `append`

**Comment by eholk** [comment]
Thanks for the change! This looks good, but I agree that adding tests would be good.

I think one test per function is better, although that's annoying because it requires a lot of new files. The advantage of one test per function is that it's easier to tell which function broke when the tests start failing.

Maybe we need a way to independently report failures when we want to have multiple tests per file...

**Comment by jitwit** [comment]
Yeah makes sense. I guess I was also thinking each test file takes like 2-2.5 seconds to process so it might be nice to be able to group more tests together without re-compiling the core. More flexibility in testing would definitely be nice.

Hah also writing out every test was definitely helpful here because my initial definition of odd: `(eq? 1 (mod0 x 2))` is no good for negatives because `mod0` preserves sign, for example! Adding a new file per function seems worth it for now at any rate

**Review by eholk — APPROVED** [review]

**Comment by eholk** [comment]
Thanks for adding the tests, and I'm glad they actually caught a bug!

Since the standard library is getting bigger, maybe it's worth scaling up the testing system some too. I opened #115 with some ideas.

**Pull request merged**